mkosi/__init__.py

```diff
@@ -6495,6 +6494,11 @@ def load_args(args: argparse.Namespace) -> MkosiConfig:
     else:
         args.environment = {}
 
+    if args.image_id is not None:
+        args.environment['IMAGE_ID'] = args.image_id
+    if args.image_version is not None:
+        args.environment['IMAGE_VERSION'] = args.image_version
```

Comment:
These should not be set in `args.environment`, since then they'll end up in `config.environment`. We only want user config in `config.environment`, not the extra stuff we determine ourselves, so this can't be added to `args.environment`, you need to make sure it only exists in `state.environment` and not in `config.environment`

Reply:
I'm not sure how you would like this resolved. I could essentially turn that `init_env()` method I wrote before into something that wraps `config.environment` during `MkosiState` instantiation, but I'd like some clarification. By this I mean:

```
state = MkosiState(
    ...
    environment = adjust_env(config),
    ...
)
```

Reply:
I could also use getters and setters, but it's getting a bit ugly.